middleware/http/opa/opa_middleware.go

```diff
@@ -0,0 +1,178 @@
+package opa
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/dapr/components-contrib/middleware"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/valyala/fasthttp"
+)
+
+type opaMiddlewareMetadata struct {
+	Rego            string `json:"rego"`
+	IncludedHeaders string `json:"includedHeaders"`
```

Comment:
Can this be array instead of what appears to be some sort of delimited list?

I think that has the bonus of grouping the `strings.Split(...)` call on line #68 closer to the parsing logic.

Reply:
Is that something that the middleware metadata will allow? If so, I'll make this swap, I'd much rather have it be an array.

Reply:
Metadata for components only supports key/Val's, so this does in fact need to be a delimited string..

Reply:
Even if the metadata value is a string, when parsing it we can use array in this intermediate structure (`opaMiddlewareMetadata`) so that no consumer of such data structure need to do the string split.

Reply:
That's possible, but basically there are no other consumers of the metadata struct, it's internal to the Middleware which gets registered in the core Dapr runtime with an interface.